### DISCUSSION TITLE
Recommended Procedure to Install Legacy Version of Zephyr and West Toolchain

### DISCUSSION THREAD

**Discussion by bob2oneil** [zephyrproject-rtos/zephyr#53847] [General] [answered]
I am building an application that has been in development for various versions of Zephyr for over a year.

This latest versions is identified by the following banner, while is compiles for the Native-POSIX target, crashes and burns badly during
execution.  I am unable to launch my **main()** C++ application anymore.  From single stepping into the source code, it appears that the background thread which invokes **main()** in the application -- seems to blow up during C++ initialization for reasons unknown.  This has not occurred for Zephyr releases dating back to Dec of 2023.

I generally follow the described install procedure which utilizes the python based 'west' to install Zephyr proper.

Assuming I wanted to effectively do a rollback to a version which did not crash and burn in Native-POSIX, what is the recommended procedure?

Is it as simple as making a legacy branch checked out in GIT, and performing a manual copy over the existing /zephyrproject/zephyr folder, or would that mix legacy rollback and later release components resulting in indeterminate behaviors? 

What is the recommended procedure to do a complete rollback to say a Dec 2022 release?

 **-- Zephyr version: 3.2.99 (/zephyrproject/zephyr), build: zephyr-v3.2.0-3473-g9d2fe69b4efc**

**Comment by yashi** [comment, marked as answer]
```
cd zephyr
git checkout v3.2.0
git show -s --pretty=format:%cd
west update
```

Make sure to run `west update` after each checkout.  This command updates/rollbacks [Zephyr Modules](https://docs.zephyrproject.org/latest/develop/modules.html) at the time of the commit.

If v3.2 works, try [bisecting](https://git-scm.com/docs/git-bisect) it to find which commit is causing the crash you are seeing.

**Reply by bob2oneil** [reply]
Many thanks
